core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -769,16 +769,22 @@ private static void planFilesFor(TableScan tableScan, String planId, int tasksPe
         Iterables.partition(tableScan.planFiles(), tasksPerPlanTask);
     int planTaskSequence = 0;
     String previousPlanTask = null;
+    String planTaskKeyPrefix = planId + "-" + tableScan.table().uuid() + "-";
```

Comment:
I think I'd fix this a bit differently with a refactor. I think we can just make `planFilesFor` itself return the initialPlanTasks + next task Pair rather than just be void. Then I think we'd be able to get rid of this "populate and later find the initial based on the seq via `initialScanTasksFor`" logic. 

In this refactored method, there'd be a short circuit logic which looks something like

```
if (Iterables.isEmpty(taskGroupings)) {
   return Pair.of(Lists.empty(), null));
}

// The refactored logic so we keep track of the initial tasks and then later return them
```

Reply:
let me know if that makes sense or not, something I was thinking through (and should've done to begin with in hindsight)

Reply:
SG to me, also if there are not next taskPair we can avoid the empty plan-tasks to be returned which is note required in this case effectively the response would be : 

```
{"status":"completed","plan-id":"sync-a1d448e2-c976-4bca-9a02-92a106e535ae","file-scan-tasks":[]}
```